clang/lib/Basic/Targets.cpp

```diff
@@ -243,6 +247,8 @@ std::unique_ptr<TargetInfo> AllocateTarget(const llvm::Triple &Triple,
   case llvm::Triple::thumbeb:
```

Comment:
github won't let me add a comment in the right spot, but I think we need:

```
  case llvm::Triple::arm:
  case llvm::Triple::thumb:
    if (Triple.isOSBinFormatMachO())
      return std::make_unique<DarwinARMTargetInfo>(Triple, Opts);
    else if (Triple.isAppleMachO())
      return std::make_unique<AppleMachOARMTargetInfo>(Triple, Opts);
  ```

Reply:
We would never hit the `else` case, `isAppleMachO()` is `(getVendor() == Triple::Apple) && isOSBinFormatMachO()`. So if `isOSBinFormatMachO()` is false, then `isAppleMachO()` will also be false.

Reply:
oh, I missed that we're already lumping `arm-none-macho` & friends under `DarwinARMTargetInfo`.

Reply:
Yeah, that's why I was asking if we should just always use Darwin for MachO instead of inconsistently using Darwin for some archs and was-AArch64leTargetInfo-now-AppleMachO for others. Keeping it inconsistent is probably the smaller/safer change, but the inconsistency does bother me.